src/hotspot/share/oops/arrayKlass.hpp

```diff
@@ -118,10 +118,6 @@ class ArrayKlass: public Klass {
   // Return a handle.
   static void     complete_create_array_klass(ArrayKlass* k, Klass* super_klass, ModuleEntry* module, TRAPS);
 
-
-  // jvm support
-  jint compute_modifier_flags() const;
```

Comment:
I would have expected this to be defined as a virtual function here and then overridden as needed by the various subclasses.

Reply:
I moved the TypeArrayKlass modifiers to apply to TypeArrayKlass, so they're not accidentally used for ObjArrayKlass, or any other new ArrayKlass that should have different modifiers.  I made the virtual function be defined for the most specific class.

Reply:
To be clear I would expect arrayKlass to define a pure virtual function for this, and then each subclass overrides as required. Otherwise you can't generally operate on an arrayKlass but must always know what subtype you are dealing with.

Reply:
Klass has a pure virtual function for this, so ArrayKlass is abstract.  There isn't a reason to redeclare it in ArrayKlass.

     virtual jint compute_modifier_flags() const = 0;

Reply:
Sorry I didn't realize that.